### ISSUE TITLE
"react/destructuring-assignment": should NOT have additional properties

### ISSUE THREAD

**Issue by khinshankhan** [jsx-eslint/eslint-plugin-react#3286]
Hello,
I'm encountering the error
```
Oops! Something went wrong! :(

ESLint: 8.12.0

Error: .eslintrc.js#overrides[4]:
	Configuration for rule "react/destructuring-assignment" is invalid:
	Value {"ignoreClassFields":false,"destructureInSignature":"always"} should NOT have additional properties.
```
with the rule
```js
        "react/destructuring-assignment": [
          `warn`,
          `always`,
          { ignoreClassFields: false, destructureInSignature: `always` },
        ],
```

I think it's the `destructureInSignature` property causing issues. Not sure why, lifted it off of the [docs](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/destructuring-assignment.md) and subbed in values. I don't think there are any typos nor conflicting rules.

Additional information:
My eslint packages are:
```js
    "eslint": "^8.12.0",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-graphql": "^4.0.0",
    "eslint-plugin-import": "^2.25.4",
    "eslint-plugin-jsx-a11y": "^6.5.1",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react": "^7.29.4",
    "eslint-plugin-react-hooks": "^4.3.0",
```

**Comment by ljharb** [comment]
Because it’s not published yet. You need to look at the docs on the latest tagged version, not on the default branch (for every repo on github).

**Comment by khinshankhan** [comment]
Oh, I see. How do we view the docs on the latest tagged version?
